browser/lib/transport/xhrrequest.js

```diff
@@ -115,7 +115,7 @@ var XHRRequest = (function() {
 
 		if(body) {
 			var contentType = headers['content-type'] || (headers['content-type'] = 'application/json');
-			if(contentType == 'application/json' && typeof(body) != 'string')
```

Comment:
Ok, well you if you go up to line 112 you see the same type of check that someone at ably did.   Figured this spot needed it too...